src/components/ProgressBar/ProgressBar.styles.tsx

```diff
@@ -9,9 +8,15 @@ export const Track = styled.div`
   overflow: hidden;
 `;
 
-export const Progress = styled(motion.div)`
+interface ProgressBarProps {
+  initialWidth: number;
+  duration: number;
+}
+
+export const Progress = styled.div<ProgressBarProps>`
   height: 100%;
-  width: 100%;
+  width: ${(props) => props.initialWidth}%;
   background-color: ${(props) => props.theme.colors.primary};
   transform-origin: left;
+  transition: width ${(props) => props.duration}s linear;
```

Comment:
could use destructuration:
`
${({duration}) =>duration}%;
`

Reply:
I haven't checked this recently, but in the past, at least, transitioning width was always less performant than transitioning `transform` and changing the `scaleX`. I think it's because transitioning width can trigger a reflow/layout wheras transforms only affect compositing.

I'm not 100% sure if this is still the case in 2022, but it's fairly simple to use `transform: scale` instead of width here, right?